float32 x
float32 y
---
int32 deltaT
---
float32 eucliDist